zulipterminal/ui_tools/boxes.py

```diff
@@ -672,30 +672,42 @@ def reactions_view(self, reactions: List[Dict[str, Any]]) -> Any:
     def footlinks_view(
         message_links: 'OrderedDict[str, Tuple[str, int, bool]]',
         footlinks_enabled: bool,
-    ) -> Any:
+        padded: bool,
+        wrap: str,
+    ) -> Tuple[Any, int]:
         # Return if footlinks are disabled by the user.
         if not footlinks_enabled:
-            return None
+            return None, 0
 
         footlinks = []
+        footlinks_width = 0
         for link, (text, index, show_footlink) in message_links.items():
             if not show_footlink:
                 continue
 
-            footlinks.extend([
+            styled_footlink = [
                 ('msg_link_index', '{}:'.format(index)),
-                ' ',
+                (None, ' '),
                 ('msg_link', link),
-                '\n',
-            ])
+            ]
+            footlinks_width = max(
+                footlinks_width,
+                sum([len(text) for style, text in styled_footlink])
+            )
+            footlinks.extend([*styled_footlink, '\n'])
 
         if not footlinks:
-            return None
+            return None, 0
 
         footlinks[-1] = footlinks[-1][:-1]  # Remove the last newline.
-        return urwid.Padding(urwid.Text(footlinks, wrap='ellipsis'),
-                             align='left', left=8, width=('relative', 100),
-                             min_width=10, right=2)
+
+        text_widget = urwid.Text(footlinks, wrap=wrap)
+        if padded:
+            return urwid.Padding(text_widget, align='left', left=8,
+                                 width=('relative', 100), min_width=10,
+                                 right=2), footlinks_width
+        else:
+            return text_widget, footlinks_width
```

Comment:
This is the only part that depends on `wrap` or `padded`? That makes it feel separate to the function, ie. the conditional belongs outside of the calling points. The padding is clearly just a wrapper, that is used in one place, and the use of the wrap parameter is less clear - it depends if we're set on returning a widget vs `footlinks`.

Reply:
This does feel out of place. We could return only the markup from `footlinks_view` (+ rename it to `markup_footlinks`) and then add wrappers as needed in the caller functions. Thoughts?